src/main/kotlin/uk/gov/justice/digital/hmpps/sentenceplan/controlller/PlanController.kt

```diff
@@ -0,0 +1,24 @@
+package uk.gov.justice.digital.hmpps.sentenceplan.controlller
+
+import org.springframework.dao.EmptyResultDataAccessException
+import org.springframework.web.bind.annotation.GetMapping
+import org.springframework.web.bind.annotation.PathVariable
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.sentenceplan.entity.PlanEntity
+import uk.gov.justice.digital.hmpps.sentenceplan.services.PlanService
+import java.util.*
+
+private const val EXPECTED_RESULT_COUNT = 1
```

Comment:
The particular Exception class I'm throwing requires an "Expected result count" value to compare with the actual result count (which defaults to zero) and which it then uses to build its message. If there's a more relevant exception I can throw and capture instead let me know and I'll change it and remove this constant.

Reply:
I did originally consider that but didn't want to overload the meaning since I thought it was being used for static asset loading, but I can't reproduce that behaviour now and so fixed in 9d03797b02e997c6fb29935e9b3772260bf23004 :)